tools/database/src/resharding.rs

```diff
@@ -0,0 +1,188 @@
+use core::time;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use near_async::time::Clock;
+use near_chain::rayon_spawner::RayonAsyncComputationSpawner;
+use near_chain::resharding::ReshardingResponse;
+use near_chain::types::ChainConfig;
+use near_chain::{Chain, ChainGenesis, DoomslugThresholdMode};
+use near_chain_configs::MutableConfigValue;
+use near_epoch_manager::shard_tracker::{ShardTracker, TrackedConfig};
+use near_epoch_manager::EpochManager;
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::types::EpochId;
+use near_primitives::types::{BlockHeight, ShardId};
+use near_store::db::{MixedDB, ReadOrder, RecoveryDB, RocksDB, SplitDB};
+use near_store::genesis::initialize_sharded_genesis_state;
+use near_store::{Mode, NodeStorage, Store, Temperature};
+use nearcore::NightshadeRuntimeExt;
+use nearcore::{open_storage, NearConfig, NightshadeRuntime};
+
+#[derive(clap::Args)]
+pub(crate) struct ReshardingCommand {
+    /// The block height at which resharding is performed.
+    /// This should be, usually, the block before shard layout has changed.
```

Comment:
Can you figure out and comment resharding is done on the pre or post state root? I think it would make more sense to do it on the post state root because then we don't need to apply the last block.

Reply:
Post state root, I'll add a note here!